internal/api/impl/auth/userinfo.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import "github.com/zitadel/oidc/v3/pkg/oidc"
+
+// UserInfoProfile is a simplified version of the oidc.UserInfoProfile structure.
+// It's been created as we want only a limited amount of user information in Perses and some data like locale can be
+// provided wrongly by some OIDC providers.
+type UserInfoProfile struct {
```

Comment:
don't you need to use this struct somewhere ?

Reply:
maybe we can put this PR in hold until this refactoring is finished, wdyt ?

Reply:
say hello to the new interface :) I'll resolve this discussion and let you review the changes as it is quite different now.